internal/idle/idle.go

```diff
@@ -93,52 +76,51 @@ type manager struct {
 type ManagerOptions struct {
 	Enforcer Enforcer
 	Timeout  time.Duration
-	Logger   grpclog.LoggerV2
 }
 
 // NewManager creates a new idleness manager implementation for the
-// given idle timeout.
-func NewManager(opts ManagerOptions) Manager {
-	if opts.Timeout == 0 {
-		return noopManager{}
+// given idle timeout.  It begins in idle mode.
+func NewManager(opts ManagerOptions) *Manager {
+	return &Manager{
+		enforcer:         opts.Enforcer,
+		timeout:          opts.Timeout,
+		actuallyIdle:     true,
+		activeCallsCount: -math.MaxInt32,
 	}
-
-	m := &manager{
-		enforcer: opts.Enforcer,
-		timeout:  int64(opts.Timeout),
-		logger:   opts.Logger,
-	}
-	m.timer = timeAfterFunc(opts.Timeout, m.handleIdleTimeout)
-	return m
 }
 
 // resetIdleTimer resets the idle timer to the given duration. This method
 // should only be called from the timer callback.
-func (m *manager) resetIdleTimer(d time.Duration) {
-	m.idleMu.Lock()
-	defer m.idleMu.Unlock()
-
-	if m.timer == nil {
-		// Only close sets timer to nil. We are done.
+func (m *Manager) resetIdleTimerLocked(d time.Duration) {
+	if m.isClosed() || m.timeout == 0 {
 		return
 	}
 
 	// It is safe to ignore the return value from Reset() because this method is
 	// only ever called from the timer callback, which means the timer has
 	// already fired.
-	m.timer.Reset(d)
+	if m.timer != nil {
+		m.timer.Stop()
+	}
```

Comment:
Couple of things here:
- The comment at the top of this block still talks about `Reset()`, which is not called anymore
- Why do we need to `Stop` the timer instead of calling `Reset()`?
  - this method is only ever called from the timer callback or when we are exiting idle. In the former case, the timer has already fired and in the latter case, the timer is not running. Can we get away with not doing `Reset` or `Stop` and simply creating a new timer here? That should simplify the code and get rid of the comment.

Reply:
> We always need to call `Stop` to prevent a timer leak

I know for a fact that we need to `Stop` a `Ticker` to prevent a leak, but i thought that wasn't the case with a `Timer`.